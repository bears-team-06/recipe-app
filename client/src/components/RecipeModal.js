import React from 'react'

const RecipeModal = (props) => {
    return (
        <div>
            <div><input type={"text"} placeholder={"Enter Recipe Name"} /></div>
            <div><input type={"textarea"} placeholder={"Enter ingredients"} /></div>
            <div><input type={"textarea"} placeholder={"Enter instructions"} /></div>
            <div><button onClick={props.closeModal}>Close</button></div>
        </div>
    )
}

export default RecipeModal