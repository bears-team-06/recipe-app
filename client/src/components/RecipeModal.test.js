import React from 'react'
import {shallow} from 'enzyme'
import RecipeModal from "./RecipeModal";
describe("Recipe Modal", () => {
  it("matches snapshot", () => {
      const RecipeModalWrapper = shallow(<RecipeModal/>)
      expect(RecipeModalWrapper).toMatchSnapshot()
  });

  it("calls callback on clicking button", () => {
      const closeModalSpy = jest.fn()
      const RecipeModalWrapper = shallow(<RecipeModal closeModal={closeModalSpy}/>)
      RecipeModalWrapper.find('button').simulate('click')
      expect(closeModalSpy).toHaveBeenCalledTimes(1)
  });
});
