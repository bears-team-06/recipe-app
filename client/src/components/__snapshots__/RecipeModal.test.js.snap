// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Modal matches snapshot 1`] = `
<div>
  <div>
    <input
      placeholder="Enter Recipe Name"
      type="text"
    />
  </div>
  <div>
    <input
      placeholder="Enter ingredients"
      type="textarea"
    />
  </div>
  <div>
    <input
      placeholder="Enter instructions"
      type="textarea"
    />
  </div>
  <div>
    <button>
      Close
    </button>
  </div>
</div>
`;
